### PR TITLE
Adding more dispatcher events for the app

### PR DESCRIPTION
**Pull request by sharidas**
Adding more dispatcher events to handle:
a) Rename of groups
b) Members leaving groups and
c) Role change of members in the group

Signed-off-by: Sujith H <sharidasan@owncloud.com>

### CODE REVIEW

File: lib/Service/MembershipHelper.php [hunk 1]
```diff
@@ -315,6 +315,13 @@ public function notifyUserRoleChange($targetUserId, array $groupInfo, array $mem
 			->setUser($targetUserId)
 			->setLink($link);
 		$this->notificationManager->notify($notification);
+		if($memberInfo['role'] === 0) {
```
Comment (PVince81) [inline lib/Service/MembershipHelper.php]: Use `Roles::BACKEND_ROLE_MEMBER` and `Roles::BACKEND_ROLE_ADMIN`

File: lib/Service/MembershipHelper.php [hunk 2]
```diff
@@ -315,6 +315,13 @@ public function notifyUserRoleChange($targetUserId, array $groupInfo, array $mem
...
+			$roleName = "Member";
+		} elseif ($memberInfo['role'] === 1) {
+			$roleName = "Group owner";
+		}
+		$event = new GenericEvent(null, ['user' => $targetUserId, 'groupName' => $groupInfo['display_name'], 'role' => $roleName]);
```
Comment (PVince81) [inline lib/Service/MembershipHelper.php]: let's provide both the numeric role and a `roleDisplayName` value for the user-friendly version, just in case

**Comment by sharidas** [comment]
- [x] Used `Roles::BACKEND_ROLE_MEMBER` and `Roles::BACKEND_ROLE_ADMIN` 
- [x] Used the numeric role along with the `roleDisplayName`

File: lib/Service/MembershipHelper.php [hunk 3]
```diff
@@ -315,6 +316,13 @@ public function notifyUserRoleChange($targetUserId, array $groupInfo, array $mem
 			->setUser($targetUserId)
 			->setLink($link);
 		$this->notificationManager->notify($notification);
+		if($memberInfo['role'] === Roles::BACKEND_ROLE_MEMBER) {
+			$roleName = "Member";
+		} elseif ($memberInfo['role'] === Roles::BACKEND_ROLE_ADMIN) {
+			$roleName = "Group owner";
+		}
+		$event = new GenericEvent(null, ['user' => $targetUserId, 'groupName' => $groupInfo['display_name'], 'roleNumber' => $memberInfo['role'], 'roleDisaplayName' => $roleName]);
```
Comment (sharidas) [inline lib/Service/MembershipHelper.php]: Both `roleDisplayName` as well as numeric roles are used.

**Pull request merged**